### PR TITLE
changed onSelect behavior

### PR DESCRIPTION
**Pull request by jquense**
I kept the changes a localized to the dropdown as possible, this was actually helpful just in that it uncovered a few focus bugs (and made some tests simpler).

changed: onRequestClose -> onToggle

onSelect passes the eventKey directly instead of the selection event.

### CODE REVIEW

**Comment by jquense** [comment]
anyone have any feels about this one?

File: src/DropdownBase.js [hunk 1]
```diff
@@ -288,27 +305,80 @@ export function menuWithMenuItemSiblings(...nonMenuTypes) {
...
+  /**
+   * Whether or not the Dropdown is visible.
+   *
+   * @controllable onToggle
+   */
   open: React.PropTypes.bool,
 
-  onRequestClose: React.PropTypes.func,
+  /**
+   * A callback fired when the Dropdown wishs to change visibility. Called with the requested
```
Comment (taion) [inline src/DropdownBase.js]: wishs -> wishes

**Comment by taion** [comment]
I like it. Can we put `uncontrollable` on `DropdownBase` though, or does it have to go on the concrete components?

Also, I wonder if `uncontrollable` could be used as a decorator... heheheh.

**Comment by jquense** [comment]
Given the current setup it can't go on the Base b/c the other components would then be exending the wrapping HoC not the actual dropdown base.

We can totally make it a decorator, i had been meaning to make that work.

**Comment by taion** [comment]
Aah, got it - that's a little annoying though, that anybody who extends this on their own would have to separately use the controller.

If we go with your proposal where we just have a `Dropdown` component with the full children-based syntax, and just keep the special-cased ones as wrappers for `Dropdown`, would we still need each of those to be wrapped with `uncontrollable`?

**Comment by jquense** [comment]
I am almost about to PR that proposal, but no you only need to wrap the base Dropdown component in that case

**Comment by taion** [comment]
Sweet. Is that on top of this? i.e. do we need to merge this first?

**Comment by jquense** [comment]
yup, merging this first would be great

**Pull request merged**